hightime/_datetime.py

```diff
@@ -52,8 +52,30 @@ def __new__impl__(
         femtosecond=0,
         yoctosecond=0,
         tzinfo=None,
-        **kwargs
+        **kwargs,
     ):
+        if (
```

Comment:
This is an extended version of https://github.com/python/cpython/blob/main/Lib/_pydatetime.py#L1763
and https://github.com/python/cpython/blob/main/Lib/_pydatetime.py#L2355

`datetime.__reduce__()` has used a 10-byte buffer with this format since Python 2.x (if not earlier).